tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs

```diff
@@ -28,6 +28,35 @@ internal class WafConfigurator
 
         public WafConfigurator(WafLibraryInvoker wafLibraryInvoker) => _wafLibraryInvoker = wafLibraryInvoker;
 
+        public bool CheckVersionCompatibility()
+        {
+            var versionWaf = _wafLibraryInvoker.GetVersion();
+            var versionWafSplit = versionWaf.Split('.');
+            if (versionWafSplit.Length != 3)
+            {
+                Log.Warning("Waf version {WafVersion} has a non expected format", versionWaf);
+                return false;
+            }
+
+            var canParse = int.TryParse(versionWafSplit[1], out var wafMinor);
+            canParse &= int.TryParse(versionWafSplit[0], out var wafMajor);
+            var tracerVersion = GetType().Assembly.GetName().Version;
+            if (tracerVersion is null || !canParse)
+            {
+                Log.Warning("Waf version {WafVersion} or tracer version {TracerVersion} have a non expected format", versionWaf, tracerVersion);
+                return false;
+            }
```

Comment:
Not a blocker, but did you consider using `System.Version` to parse this?

Reply:
Good point. It's a 3 digits version, don't know how it'd behave with it. But I note it for another PR.